visual_behavior/data_access/reformat.py

```diff
@@ -49,6 +49,41 @@ def add_session_type_exposure_number_to_experiments_table(experiments):
     return experiments
 
 
+def add_engagement_state_to_trials_table(trials, extended_stimulus_presentations):
+    '''
+    adds `engaged` and `engagement_state` fields to the trial table
+    both are pulled from the value of the stimulus table that is closest to the time at the start of each trial
+    '''
+    extended_stimulus_presentations = extended_stimulus_presentations
+
+    # for each trial, find the stimulus index that is closest to the trial start
+    # add to a new column called 'first_stim_presentation_index'
+    for idx, trial in trials.iterrows():
+        start_time = trial['start_time']
+        query_string = 'start_time > @start_time - 1 and start_time < @start_time + 1'
+        first_stim_presentation_index = np.argmin(
+            np.abs(
+                start_time - extended_stimulus_presentations.query(query_string)['start_time']
+            )
+        )
+        trials.at[idx, 'first_stim_presentation_index'] = first_stim_presentation_index
+
+    # define the columns from extended_stimulus_presentations that we want to merge into trials
+    cols_to_merge = [
```

Comment:
@farznaj see this code block. If you want access to more attributes from the extended stimulus presentations, this is the place to add them.

Reply:
Right now, I'm only merging in 'engaged' and 'engagement_state'

Reply:
@doug looks good to me; lets go with engagement for now, and if something came up later we can add it.